sbtc/src/deposits.rs

```diff
@@ -93,19 +97,51 @@ pub struct ParsedDepositRequest {
     pub lock_time: u64,
 }
 
+/// A full deposit, with a fully extracted and verified `scriptPubKey`
+#[derive(Debug, Clone)]
+pub struct Deposit {
+    /// The transaction spent to the signers as a deposit for sBTC.
+    pub tx: Transaction,
+    /// The deposit information included in one of the output
+    /// `scriptPubKey`s of the above transaction.
+    pub info: DepositInfo,
+    /// The block hash of the block that includes this transaction. If this
+    /// is `None` then this transaction is in the mempool. TODO(384): In
+    /// the case of a reorg, it's not clear what happens if this was
+    /// confirmed.
+    pub block_hash: Option<BlockHash>,
+    /// The number of confirmations deep from that chain tip of the bitcoin
+    /// block that includes this transaction. If `None` then this is in the
+    /// mempool. TODO(384): In the case of a reorg, it's not clear what
+    /// happens if this was confirmed.
+    pub confirmations: Option<u32>,
+    /// This is to make sure that this struct cannot be created without the
+    /// above invariants being upheld.
```

Comment:
Oh you're right it doesn't uphold them, it just makes sure that the struct cannot be created (outside of this file) without the invariants being upheld, since you cannot set a private field of a struct outside of the file that creates it.

Reply:
Ah that makes waaay more sense 😂